src/qibolab/instruments/qm.py

```diff
@@ -597,28 +619,59 @@ def declare_output(self, options, threshold=None, angle=None):
             raise_error(ValueError, f"Invalid acquisition type {acquisition_type}.")
         self.acquisition.assign_element(self.element)
 
-    def bake(self, config: QMConfig):
-        if self.baked is not None:
-            raise_error(RuntimeError, f"Bake was already called for {self.pulse}.")
-        # ! Only works for flux pulses that have zero Q waveform
-
-        # Create the different baked sequences, each one corresponding to a different truncated duration
-        # for t in range(self.pulse.duration + 1):
-        with baking(config.__dict__, padding_method="right") as self.baked:
-            # if t == 0:  # Otherwise, the baking will be empty and will not be created
-            #    wf = [0.0] * 16
-            # else:
-            # wf = waveform[:t].tolist()
-            if self.pulse.duration == 0:
-                waveform = [0.0] * 16
-            else:
-                waveform = self.pulse.envelope_waveform_i.data.tolist()
-            self.baked.add_op(self.pulse.serial, self.element, waveform)
-            self.baked.play(self.pulse.serial, self.element)
-            # Append the baking object in the list to call it from the QUA program
-            # self.segments.append(b)
 
-        self.duration = self.baked.get_op_length()
+DurationsType = Union[List[int], np.ndarray]
+
+
+@dataclass
+class BakedPulse(QMPulse):
+    """Baking allows 1ns resolution in the pulse waveforms."""
+
+    segments: List[Baking] = field(default_factory=list)
+    """Baked segments implementing the pulse."""
+    amplitude: Optional[float] = None
+    """Amplitude of the baked pulse. Relevant only when sweeping amplitude."""
+    durations: Optional[DurationsType] = None
+
+    def __hash__(self):
+        return super().__hash__()
```

Comment:
I think it is different, because the `@dataclass` decorator acts after the inheritance and overwrites the default `__hash__`. If I remove it I get an error that `BakedPulse` is unhashable.

Reply:
I see...

I thought it wouldn't have erased the inherited one:

> By default, [dataclass()](https://docs.python.org/3/library/dataclasses.html#dataclasses.dataclass) will not implicitly add a [__hash__()](https://docs.python.org/3/reference/datamodel.html#object.__hash__) method unless it is safe to do so. Neither will it add or change an existing explicitly defined [__hash__()](https://docs.python.org/3/reference/datamodel.html#object.__hash__) method. Setting the class attribute __hash__ = None has a specific meaning to Python, as described in the [__hash__()](https://docs.python.org/3/reference/datamodel.html#object.__hash__) documentation.

> If [__hash__()](https://docs.python.org/3/reference/datamodel.html#object.__hash__) is not explicitly defined, or if it is set to None, then [dataclass()](https://docs.python.org/3/library/dataclasses.html#dataclasses.dataclass) may add an implicit [__hash__()](https://docs.python.org/3/reference/datamodel.html#object.__hash__) method. Although not recommended, you can force [dataclass()](https://docs.python.org/3/library/dataclasses.html#dataclasses.dataclass) to create a [__hash__()](https://docs.python.org/3/reference/datamodel.html#object.__hash__) method with unsafe_hash=True. This might be the case if your class is logically immutable but can nonetheless be mutated. This is a specialized use case and should be considered carefully.

But apparently it's not taking into account inheritance...